### DISCUSSION TITLE
Jotai's selectorFamily creating explaination 🙏🙏

### DISCUSSION THREAD

**Discussion by kaustubh2525** [pmndrs/jotai#2835] [Q&A] [answered]
I just wanna know the explaination on how would i create a selectorFamily in jotai in such a way just like recoil's selectorFamily which takes in the args and specifically derives the piece of data / state for some usage with a small example ??

**Comment by dai-shi** [comment, marked as answer]
Jotai's atomFamily has a different signature from Recoil's one and should cover both atomFamily and selectorFamily use cases of Recoil. It's not the same, and there might be some capabilities missing.

```js
import { atomFamily } from 'jotai/utils';

const baseAtom = atom({});
const myFamily = atomFamily((key) => atom((get) => get(baseAtom)[key]));
```

FYI, Jotai's atomFamily implementation is pretty simple. See this simplified code:
https://github.com/pmndrs/jotai/discussions/1323#discussioncomment-3303035
```js
// simplified code
const atomFamily = (initializeAtom) => {
  const cache = new Map();
  return (param) => {
    if (!cache.has(param)) {
      cache.set(param, initializeAtom(param));
    }
    return cache.get(param);
  };
};
```
